src/file/file.service.ts

```diff
@@ -68,9 +69,44 @@ export class FileService {
     if (createdFile) {
       await QuotaService.updateUsed(ownerID, size);
     }
+    if(this.isShortcut(createdFile)) createdFile.isShortcut = true;
+    else createdFile.isShortcut = false;
+
     return createdFile;
   }
 
+
+  /**
+   * Creates a shortcut file and adds it to the DB.
+   * @param partialFile - a partial file containing some of the fields.
+   * @param name - the name of the file with the extensions.
+   * @param size - the size of the file.
+   * @param fileId - the id of the original file.
+   * @param parent - the id of the folder in which the file will reside (in the GUI).
+   */
+
+  public static async createShortcut(
+    name: string,
+    fileID: string,
+    size: number = 0, parent: string): Promise<IFile> {
+
+    // basicFile is without key and bucket - in case it is a folder.
+    let shortcutFile = {
+      name,
+      fileID,
+      size,
+      parent
+    };
+
+    const file: IShortcut = new shortcutModel(shortcutFile);
+    const createdShortcut: IFile = await FilesRepository.createShortcut(file);
+
+    if (this.isShortcut(createdShortcut)) createdShortcut.isShortcut = true
+    else createdShortcut.isShortcut = false;
```

Comment:
```suggestion
    createdShortcut.isShortcut = this.isShortcut(createdShortcut);
```

Reply:
This occurs in many places

Reply:
you mean its redundant or incorrect?

Reply:
it cannot work(i tried), because the isShortcut method isnt assigned to the type of the isShortcut parameter at the primitiveFIle class. plus when i assign it correctly like vs-code told me to, it made more issues and i couldnt manage to make it work.

Reply:
and i know you're right, its just not working

Reply:
done